src/lsdb/core/plotting/skymap.py

```diff
@@ -1,32 +1,61 @@
-from typing import Any, Dict
+from __future__ import annotations
+
+from typing import Any, Callable, Dict
 
 import healpy as hp
 import numpy as np
+import pandas as pd
+from dask import delayed
 from hipscat.pixel_math import HealpixPixel
 
+from lsdb.dask.merge_catalog_functions import filter_by_hipscat_index_to_pixel
+
+
+@delayed
+def perform_inner_skymap(
+    partition: pd.DataFrame,
+    func: Callable[[pd.DataFrame, HealpixPixel], Any],
+    pixel: HealpixPixel,
+    target_order: int,
+    **kwargs,
+) -> np.ndarray:
+    """Splits a partition into pixels at a target order and performs a given function on the new pixels"""
+    delta_order = target_order - pixel.order
+    pixels = np.arange(pixel.pixel << (2 * delta_order), (pixel.pixel + 1) << (2 * delta_order))
+    return np.vectorize(
+        lambda p: func(
+            filter_by_hipscat_index_to_pixel(partition, target_order, p),
+            HealpixPixel(target_order, p),
+            **kwargs,
+        )
+    )(pixels)
+
 
-def plot_skymap(pixel_map: Dict[HealpixPixel, Any], **kwargs):
-    """Plot a map of healpix_pixels with values on a mollweide projection.
+def compute_skymap(pixel_map: Dict[HealpixPixel, Any], order: int | None = None, default_value: Any = 0.0):
+    """Returns a histogram map of healpix_pixels to values.
 
     Args:
         pixel_map(Dict[HealpixPixel, Any]): A dictionary of healpix pixels and their values
-        kwargs: Additional keyword arguments to pass to `healpy.mollview`
+        order (int): The order to make the histogram at
+        default_value: The value to use at pixels that aren't covered by the pixel_map
     """
 
     pixels = list(pixel_map.keys())
     hp_orders = np.vectorize(lambda x: x.order)(pixels)
     hp_pixels = np.vectorize(lambda x: x.pixel)(pixels)
-    max_order = np.max(hp_orders)
-    npix = hp.order2npix(max_order)
-    img = np.zeros(npix)
-    dorders = max_order - hp_orders
-    values = np.vectorize(lambda x: pixel_map[x])(pixels)
-    starts = hp_pixels * (4**dorders)
-    ends = (hp_pixels + 1) * (4**dorders)
+    if order is None:
+        order = np.max(hp_orders)
+    npix = hp.order2npix(order)
+    img = np.full(npix, default_value)
+    dorders = order - hp_orders
+    values = [pixel_map[x] for x in pixels]
+    starts = hp_pixels << (2 * dorders)
+    ends = (hp_pixels + 1) << (2 * dorders)
 
     def set_values(start, end, value):
         img[np.arange(start, end)] = value
 
-    np.vectorize(set_values)(starts, ends, values)
+    for s, e, v in zip(starts, ends, values):
```

Comment:
I've seen mixed things on this, but I think the answer is that for ufuncs like set_values it basically just wraps the for loop anyway. I haven't profiled to see if there's a performance impact. The reason this is a for loop is that np.vectorize doesn't work for when values is a list of arrays of different dimensions, which it will be when a fixed order is set.